app/controllers/api/v1/authentication_controller.rb

```diff
@@ -29,27 +31,33 @@ def signup
 
   # POST api/v1/oauth/login
   def oauth_login
-    @user = User.find_by!(email: params["email"])
+    @user = User.find_by!(email: @response["email"])
     token = JsonWebToken.encode(
       user_id: @user.id, username: @user.name, email: @user.email
     )
     render json: { token: token }, status: :accepted
   end
 
   # POST api/v1/oauth/signup
+  # rubocop:disable Metrics/AbcSize
   def oauth_signup
-    if User.exists?(email: params[:email])
+    if User.exists?(email: @response["email"])
       api_error(status: 409, errors: "user already exists")
     else
-      @user = User.new(oauth_signup_params)
-      @user.password = Devise.friendly_token[0, 20]
-      @user.save!
+      @user = User.create!(
+        name: @response["name"],
+        email: @response["email"],
+        password: Devise.friendly_token[0, 20],
+        provider: params[:provider],
+        uid: @response["id"] || @response["sub"]
+      )
```

Comment:
There can be scenarios like empty email from the response, for instance you can create a Facebook account with email linked to it. So handle failing to create user record with proper error message

Reply:
@tachyons can you please check once!!